datacite-update-doi-handler/src/main/java/no/unit/nva/datacite/handlers/UpdateDoiEventHandler.java

```diff
@@ -85,7 +92,14 @@ private void makeDoiRegisteredIfPublicationGone(
 
         if (e.getStatus().getStatusCode() == GONE.getStatusCode()) {
             logger.info(SHOULD_REMOVE_METADATA_LOG_MESSAGE, input.getPublicationId());
-            deleteMetadata(input.getCustomerId(), doi);
+
+            var resource = getMetadata(input, doi);
```

Comment:
Should getMetadata return optional since it may be null? It can then be mapped…

Reply:
not sure "it can" be null? but duplicateOf can and could be an optional